lib/connection-manager.js

```diff
@@ -60,145 +79,68 @@ module.exports.connectLogServer = function(config, programName) {
             , tailmode: config.TAIL_MODE !== 'trs' ? config.TAIL_MODE : undefined
             , transport: config.TRANSPORT !== 'http' ? config.TRANSPORT : undefined
         }
-        , reconnection: config.LOGDNA_RECONNECT
-        , perMessageDeflate: !!config.COMPRESS
     };
+    var firstrun = true;
     var proxy = getProxyFromURI(url.parse(logurl));
     if (proxy) options.agent = new HttpsProxyAgent(url.parse(proxy));
-
-    var firstrun = true;
     debug(logurl);
-    var socket = new WebSocket(logurl, options);
-    linebuffer.setSocket(socket);
-    debug(socket.options);
-    return new Promise(resolve => {
-        debug('connecting to websocket');
-        socket.on('open', function() {
-            log('Connected to ' + config.LOGDNA_LOGHOST + ':' + config.LOGDNA_LOGPORT + ' (' + socket._socket.remoteAddress + ')' + (config.LOGDNA_LOGSSL ? ' (SSL)' : '') + (proxy ? ' via ' + proxy : ''));
-
-            if (firstrun) {
-                firstrun = false;
-
-                // start streaming logs from logdir(s) on startup
-                fileUtils.streamAllLogs(config);
-
-                if (os.platform() === 'win32' && config.winevent) {
-                    if (typeof config.winevent === 'string') {
-                        config.winevent = config.winevent.split(',');
-                    }
-
-                    winUtils.streamEventLog({
-                        events: config.winevent
-                    }, socket);
-
+    var req = linebuffer.setConnection(logurl, options);
+    return new Promise((resolve) => {
+        setInterval(() => {
```

Comment:
I still think you will need to not run this as a `setInterval` even though it's ok to do the retry portion